jerry-core/rcs/rcs-records.h

```diff
@@ -0,0 +1,175 @@
+/* Copyright 2015 Samsung Electronics Co., Ltd.
+ * Copyright 2015 University of Szeged
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef RCS_RECORDS_H
+#define RCS_RECORDS_H
+
+#include "ecma-globals.h"
+
+#define RCS_RECORD_TYPE_FREE                     0
+#define RCS_RECORD_TYPE_CHARSET                  1
+#define RCS_RECORD_TYPE_MAGIC_STR                2
+#define RCS_RECORD_TYPE_MAGIC_STR_EX             3
+#define RCS_RECORD_TYPE_NUMBER                   4
+
+#define RCS_RECORD_TYPE_FIRST                    RCS_RECORD_TYPE_CHARSET
+#define RCS_RECORD_TYPE_LAST                     RCS_RECORD_TYPE_NUMBER
+#define RCS_RECORD_TYPE_MIN                      RCS_RECORD_TYPE_FREE
+#define RCS_RECORD_TYPE_MAX                      RCS_RECORD_TYPE_NUMBER
+
+#define RCS_RECORD_TYPE_IS_FREE(type)            (type == RCS_RECORD_TYPE_FREE)
+#define RCS_RECORD_TYPE_IS_NUMBER(type)          (type == RCS_RECORD_TYPE_NUMBER)
+#define RCS_RECORD_TYPE_IS_CHARSET(type)         (type == RCS_RECORD_TYPE_CHARSET)
+#define RCS_RECORD_TYPE_IS_MAGIC_STR(type)       (type == RCS_RECORD_TYPE_MAGIC_STR)
+#define RCS_RECORD_TYPE_IS_MAGIC_STR_EX(type)    (type == RCS_RECORD_TYPE_MAGIC_STR_EX)
+#define RCS_RECORD_TYPE_IS_VALID(type)           (type <= RCS_RECORD_TYPE_MAX)
```

Comment:
We should add parentheses around the usage of the `type` arguments:
`  ((type) <= ...`
just to make the check extra safe.